tck/features/MatchAcceptance2.feature

```diff
@@ -0,0 +1,2023 @@
+      MATCH (a)-->(b)
+      MATCH (c)-->(d)
+      RETURN a, b, c, d
+      """
+    Then the result should be:
+      | a    | b    | c    | d    |
+      | (:A) | (:B) | (:A) | (:B) |
+      | (:A) | (:B) | (:A) | (:C) |
+      | (:A) | (:C) | (:A) | (:B) |
+      | (:A) | (:C) | (:A) | (:C) |
+    And no side effects
+
+  Scenario: Non-optional matches should not return nulls
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A), (b:B {id: 1}), (c:C {id: 2}), (d:D)
+      CREATE (a)-[:T]->(b),
+             (a)-[:T]->(c),
+             (a)-[:T]->(d),
+             (b)-[:T]->(c),
+             (b)-[:T]->(d),
+             (c)-[:T]->(d)
+      """
+    When executing query:
+      """
+      MATCH (a)--(b)--(c)--(d)--(a), (b)--(d)
+      WHERE a.id = 1
+        AND c.id = 2
+      RETURN d
+      """
+    Then the result should be:
+      | d    |
+      | (:A) |
+      | (:D) |
+    And no side effects
+
+  Scenario: Handling cyclic patterns
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a {name: 'a'}), (b {name: 'b'}), (c {name: 'c'})
+      CREATE (a)-[:A]->(b),
+             (b)-[:B]->(a),
+             (b)-[:B]->(c)
+      """
+    When executing query:
+      """
+      MATCH (a)-[:A]->()-[:B]->(a)
+      RETURN a.name
+      """
+    Then the result should be:
+      | a.name |
+      | 'a'    |
+    And no side effects
+
+  Scenario: Handling cyclic patterns when broken into two parts
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a {name: 'a'}), (b {name: 'b'}), (c {name: 'c'})
+      CREATE (a)-[:A]->(b),
+             (b)-[:B]->(a),
+             (b)-[:B]->(c)
+      """
+    When executing query:
+      """
+      MATCH (a)-[:A]->(b), (b)-[:B]->(a)
+      RETURN a.name
+      """
+    Then the result should be:
+      | a.name |
+      | 'a'    |
+    And no side effects
+
+  Scenario: Handling fixed-length variable length pattern
+    Given an empty graph
+    And having executed:
+      """
+      CREATE ()-[:T]->()
+      """
+    When executing query:
+      """
+      MATCH (a)-[r*1..1]->(b)
+      RETURN r
+      """
+    Then the result should be:
+      | r      |
+      | [[:T]] |
+    And no side effects
+
+  Scenario: Matching from null nodes should find nothing, no matches
+    Given an empty graph
+    When executing query:
+      """
+      OPTIONAL MATCH (a)
+      WITH a
+      MATCH (a)-->(b)
+      RETURN b
+      """
+    Then the result should be:
+      | b |
+    And no side effects
+
+  Scenario: Matching from null nodes should find nothing, filtered out
+    Given an empty graph
+    And having executed:
+      """
+      CREATE ()-[:T]->()
+      """
+    When executing query:
+      """
+      OPTIONAL MATCH (a:Label)
+      WITH a
+      MATCH (a)-->(b)
+      RETURN b
+      """
+    Then the result should be:
+      | b |
+    And no side effects
+
+  Scenario: Optionally matching from null nodes should give null
```

Comment:
"Optionally matching from null nodes should give null" -> "Optionally matching from null nodes should return null"